### DISCUSSION TITLE
Linkerd with Apache Apisix

### DISCUSSION THREAD

**Discussion by BenB196** [linkerd/linkerd2#6541] [Q&A]
Hi all,

I was wondering if anyone has tried integrating linkerd with [Apache Apisix](https://apisix.apache.org/)?

If you have, how well did the integration work, and would you be able to provide an example of what you did to get it to work?

I currently use the "free" tier of Traefik for ingress, but looking at potentially replacing it with something more opensource.

**Comment by adleong** [comment]
Hi @BenB196!  I haven't tried Apisix specifically, but Linkerd should generally work with any choice of ingress or api gateway.  See the ingress docs for more details: https://linkerd.io/2.10/tasks/using-ingress/

**Reply by moonming** [reply]
Hi, @adleong we can make a try whether APISIX can works well with Linkerd :) I think it's will be a good start for both projects.

**Reply by benjaminmoskovitz** [reply]
Can you guys provide me feedback on your task of integrating this two

**Reply by membphis** [reply]
> I haven't tried Apisix specifically, but Linkerd should generally work with any choice of ingress or api gateway

Yes, this should be runnable. 
We'll try: APISIX Ingress + Linkerd, just to confirm they work together.

After this, then we can submit a PR to add a new option: https://linkerd.io/2.10/tasks/using-ingress/, what do you think? @tao12345666333

**Reply by tao12345666333** [reply]
In fact, I did a simple experiment last year, and they work together.

I'll try again with the latest version when I have time in the near future, and if they still work together, I can submit a PR to update the relevant documentation.

Thank you for your interest in integrating APISIX Ingress with the Linkerd project.

**Reply by benjaminmoskovitz** [reply]
I will let you know how that works

**Reply by hadican** [reply]
Hi! Any updates on trial? :)

**Reply by tao12345666333** [reply]
> Hi! Any updates on trial? :)

Maybe I can schedule it for next week

**Reply by orhunvatansever** [reply]
any updates?

**Reply by tao12345666333** [reply]
Not tested with latest version.

I checked my list, next week at the earliest, two weeks at the latest, I can have time to work on it

**Reply by tao12345666333** [reply]
I'm already testing and writing articles for the record

**Comment by lich-y** [comment]
please try this yaml
```
apiVersion: apisix.apache.org/v2beta3
kind: ApisixRoute
metadata:
  name: linkerd
  namespace: linkerd-viz
spec:
  http:
  - backends:
    - serviceName: web
      servicePort: 8084
    match:
      hosts:
      - linkerd.yoursite.com
      paths:
      - /*
    name: web
```
then you need to modify the roundrobin about linkerd-viz_web_8084.
![image](https://user-images.githubusercontent.com/51944841/196390123-a2e1e76f-0702-4c70-8431-d9b003f25580.png)
